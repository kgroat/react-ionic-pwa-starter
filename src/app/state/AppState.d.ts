
import { RouterState } from 'react-router-redux'

import { AuthState } from './AuthState'
import { Entities } from './Entities'
import { UiState } from './UiState'

export interface AppState {
  auth: AuthState
  router: RouterState
  entities: Entities
  ui: UiState
}
