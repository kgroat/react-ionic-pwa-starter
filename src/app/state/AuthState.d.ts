
export interface AuthState {
  token: string | null
}
