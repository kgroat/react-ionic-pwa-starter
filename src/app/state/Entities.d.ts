
import { User } from 'models/user'
import { FlatMap } from './FlatMap'

export interface Entities {
  users: FlatMap<User>
}
