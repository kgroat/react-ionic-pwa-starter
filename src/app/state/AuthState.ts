
export interface AuthState {
  token: string | null
}

export const DEFAULT_STATE: AuthState = {
  token: null,
}
