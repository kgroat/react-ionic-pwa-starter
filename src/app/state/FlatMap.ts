
export interface FlatMap<T> {
  [id: string]: T | undefined
}
