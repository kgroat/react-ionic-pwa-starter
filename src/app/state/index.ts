
export { AppState, DEFAULT_STATE } from './AppState'
