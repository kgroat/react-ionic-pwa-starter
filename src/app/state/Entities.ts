
import { User } from 'models/user'
import { FlatMap } from './FlatMap'

export interface Entities {
  users: FlatMap<User>
}

export const DEFAULT_STATE: Entities = {
  users: {},
}
