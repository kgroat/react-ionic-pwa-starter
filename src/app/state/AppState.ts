
import { RouterState } from 'react-router-redux'

import { AuthState, DEFAULT_STATE as auth } from './AuthState'
import { Entities, DEFAULT_STATE as entities } from './Entities'
import { UiState, DEFAULT_STATE as ui } from './UiState'

export interface AppState {
  router: RouterState
  auth: AuthState
  entities: Entities
  ui: UiState
}

export const DEFAULT_STATE: AppState = {
  router: undefined as any,
  auth,
  entities,
  ui,
}
