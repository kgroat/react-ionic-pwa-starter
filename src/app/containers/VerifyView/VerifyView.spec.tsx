
import * as React from 'react'
import { create } from 'react-test-renderer'
import { MemoryRouter, Route } from 'react-router'

import * as reduxMockStore from 'redux-mock-store'
const createMockStore: typeof reduxMockStore.default = reduxMockStore as any
import thunk from 'redux-thunk'

import { AppState, DEFAULT_STATE } from 'state'

import ConnectedVerifyView, { VerifyView } from './VerifyView'

const mockStore = createMockStore<AppState>([thunk])
jest.mock('containers/ContentView')
jest.mock('thunks/fetchUser')
jest.mock('redux-store/store')
jest.mock('redux-store/api', () => ({
  GET: () => Promise.resolve({ success: true }),
  HttpError: class extends Error {},
}))

describe(`<${VerifyView.name} />`, () => {
  describe('snapshots', () => {
    it('should render correctly', () => {
      const store = mockStore(DEFAULT_STATE)
      const tree = create((
        <MemoryRouter>
          <Route render={props => <ConnectedVerifyView store={store} {...props} />} />
        </MemoryRouter>
      ))

      expect(tree).toMatchSnapshot()
    })
  })
})
