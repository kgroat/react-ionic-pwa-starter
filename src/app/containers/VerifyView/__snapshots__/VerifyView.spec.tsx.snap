// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifyView /> snapshots should render correctly 1`] = `
<ContentView
  noVerificationWarning={true}
  title="Verify"
>
  <div
    className="validationResult "
  >
    Verifying your email...
  </div>
</ContentView>
`;
